workspaces/optic-engine/src/projections/learners/undocumented_bodies.rs

```diff
@@ -213,61 +202,84 @@ impl EndpointBody {
-        query_parameters_shape_id: None,
-      }),
-      BodyAnalysisLocation::UnmatchedResponse {
-        status_code,
         path_id,
         method,
-        ..
-      } => EndpointBody::Response(EndpointResponseBody {
-        body_descriptor,
-        path_id: path_id.clone(),
-        method: method.clone(),
-        commands: body_commands.into_iter().collect(),
-        status_code: *status_code,
-      }),
-      _ => panic!("EndpointBody should only be created for unmatched responses and requests"),
-    }
-  }
+        content_type,
+      } => {
+        let body_descriptor = root_shape_id.map(|root_shape_id| EndpointBodyDescriptor {
+          content_type: content_type
+            .as_ref()
+            .expect("root shape id implies a content type to be present")
+            .clone(),
+          root_shape_id,
+        });
 
-  fn with_query_params(
-    &mut self,
-    root_shape_id: String,
-    body_commands: impl IntoIterator<Item = SpecCommand>,
-  ) {
-    match self {
-      EndpointBody::Request(request_body) => {
-        request_body.query_parameters_shape_id = Some(root_shape_id);
-        request_body.commands.extend(body_commands);
+        EndpointBody::Request(EndpointRequestBody {
+          body_descriptor,
+          path_id: path_id.clone(),
+          method: method.clone(),
+          commands: body_commands.into_iter().collect(),
+          query_parameters_shape_id: None,
+        })
       }
-      EndpointBody::Response(_) => {
-        panic!("query parameters can only be added to the Request variant of an EndpointBody")
+      BodyAnalysisLocation::UnmatchedQueryParameters { path_id, method } => {
+        EndpointBody::QueryParameters(EndpointQueryParameters {
+          path_id: path_id.clone(),
+          method: method.clone(),
+          root_shape_id: root_shape_id.clone(),
+          commands: body_commands.into_iter().collect(),
+        })
       }
+      BodyAnalysisLocation::UnmatchedResponse {
+        status_code,
+        path_id,
+        method,
+        content_type,
+      } => {
+        let body_descriptor = root_shape_id.map(|root_shape_id| EndpointBodyDescriptor {
+          content_type: content_type
+            .as_ref()
+            .expect("root shape id implies a content type to be present")
+            .clone(),
+          root_shape_id,
+        });
+        EndpointBody::Response(EndpointResponseBody {
+          body_descriptor,
+          path_id: path_id.clone(),
+          method: method.clone(),
+          commands: body_commands.into_iter().collect(),
+          status_code: *status_code,
+        })
+      }
+      _ => panic!("EndpointBody should only be created for unmatched responses and requests"),
     }
   }
 
   fn append_endpoint_commands(&mut self, ids: &mut impl SpecIdGenerator) {
     match self {
+      EndpointBody::QueryParameters(query_parameters) => {
+        let query_params_id = ids.request(); // TODO: give query params their own prefix
```

Comment:
Extra todo? Or is this done somewhere else?

Reply:
Nice find, extra todo indeed!

Reply:
Resolved in https://github.com/opticdev/optic/pull/950/commits/1041cddb7e56381665ff6ce0e0a1d320edc68a0d